src/DotNetLightning.Core/Transactions/Transactions.fs

```diff
@@ -403,18 +408,31 @@ module Transactions =
         let lockTime = obscuredCommitmentNumber.LockTime
 
         let tx =
-            let tx = network.CreateTransaction()
-            tx.Version <- 2u
-            let txin = TxIn(inputInfo.Outpoint)
-            txin.Sequence <- sequence
-            tx.Inputs.Add(txin) |> ignore
+            let txb = createDeterministicTransactionBuilder network
+            txb.SetVersion(TxVersionNumberOfCommitmentTxs) |> ignore
+            txb.AddCoin(
+                inputInfo,
+                CoinOptions(
+                    Sequence = (Nullable sequence)
+                )
+            ) |> ignore
             let txOuts =
                 ([toLocalDelayedOutput_opt; toRemoteOutput_opt;] |> List.choose id |> List.map(fun x -> x, None))
                 @ (htlcOfferedOutputsWithMetadata)
                 @ htlcReceivedOutputsWithMetadata
-            tx.Outputs.AddRange(txOuts |> sortTxOut)
-            tx.LockTime <- lockTime
-            tx
+
+            let txb, outAmount =
+                List.fold
+                    (fun ((txb, outAmount): TransactionBuilder * Money) (txOut: TxOut) ->
+                        txb.Send(txOut.ScriptPubKey, txOut.Value) |> ignore
+                        (txb, outAmount + txOut.Value)
+                    )
+                    (txb, Money(0UL))
+                    (sortTxOut txOuts)
+            let actualFee = inputInfo.Amount - outAmount
+            txb.SendFees actualFee |> ignore
+            txb.SetLockTime lockTime |> ignore
```

Comment:
nit: chaining methods are more concise than `ignore` ing every time.
e.g.
```fs
txb
    .SendFees(actualFee)
    .SetLocktime(lockTime)
    |> ignore
```
Afterall that is what the fluent API is designed for.